src/Symfony/Component/String/AbstractString.php

```diff
@@ -0,0 +1,555 @@
+
+        return $this->pad($length, $pad, STR_PAD_LEFT);
+    }
+
+    /**
+     * @return static
+     */
+    public function padRight(int $length, string $padStr = ' '): self
+    {
+        if ('' === $padStr || !preg_match('//u', $padStr)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $pad = clone $this;
+        $pad->string = $padStr;
+
+        return $this->pad($length, $pad, STR_PAD_RIGHT);
+    }
+
+    /**
+     * @return static
+     */
+    abstract public function prepend(string $prefix): self;
+
+    /**
+     * @return static
+     */
+    public function repeat(int $multiplier): self
+    {
+        $str = clone $this;
+        $str->string = str_repeat($str->string, $multiplier < 0 ? 0 : $multiplier);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    abstract public function replace(string $from, string $to): self;
+
+    /**
+     * @param string|callable $to
+     *
+     * @return static
+     */
+    public function replaceMatches(string $from, $to): self
+    {
+        if ('g' === $pattern[-1]) {
+            $pattern = substr($pattern, 0, -1);
+        }
+
+        if ($this->ignoreCase) {
+            $pattern .= 'i';
+        }
+
+        if (\is_array($to) || $to instanceof \Closure) {
+            $to = static function (array $m) use ($to): string {
+                $to = $to($m);
+
+                if ('' !== $to && (!\is_string($to) || !preg_match('//u', $to))) {
+                    throw new InvalidArgumentException('Replace callback must return a valid UTF-8 string.');
+                }
+
+                return $to;
+            };
+        } elseif ('' !== $to && !preg_match('//u', $to)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        } else {
+            $replace = 'preg_replace';
+        }
+
+        set_error_handler(static function ($t, $m) { throw new InvalidArgumentException($m); });
+
+        try {
+            if (null === $string = $replace($from.'u', $to, $this->string)) {
+                $lastError = preg_last_error();
+
+                foreach (get_defined_constants(true)['pcre'] as $k => $v) {
+                    if ($lastError === $v && '_ERROR' === substr($k, -6)) {
+                        throw new RuntimeException('Matching failed with '.$k.'.');
+                    }
+                }
+
+                throw new RuntimeException('Matching failed with unknown error code.');
+            }
+        } finally {
+            restore_error_handler();
+        }
+
+        $str = clone $this;
+        $str->string = $string;
+
+        return $str;
+    }
+
+    /**
+     * @return static[]
+     */
+    abstract public function split(string $delimiter, int $limit = null): array;
+
+    abstract public function startsWith(string $prefix): bool;
+
+    /**
+     * @return static
+     */
+    abstract public function substr(int $start = 0, int $length = null): self;
```

Comment:
```suggestion
    abstract public function slice(int $start = 0, int $length = null): self;
```

Reply:
`str` is not need in the name, its clear that we are dealing with a string :)

Reply:
any other opinion on this?

Reply:
I prefer `substr()` because it's easy for people to remember because it already exists in PHP and is for strings. Slice makes me thing to array because of `array_slice()`.
Now, @azjezz is right as well, there is no need to have `str` in the name.
So, I don't know what's the best option here.

Reply:
let's go with `slice()` and see if this triggers opposition :)